llama-index-integrations/readers/llama-index-readers-microsoft-sharepoint/llama_index/readers/microsoft_sharepoint/base.py

```diff
@@ -116,12 +116,17 @@ def _get_access_token(self) -> str:
             data=payload,
         )
 
-        if response.status_code == 200 and "access_token" in response.json():
-            return response.json()["access_token"]
+        json_response = response.json()
+
+        if response.status_code == 200 and "access_token" in json_response:
+            return json_response["access_token"]
 
         else:
-            logger.error(response.json()["error"])
-            raise ValueError(response.json()["error_description"])
+            error_message = json_response.get("error_description") or json_response.get(
+                "error"
```

Comment:
Should we default to something a little more descriptive? "Unknown error retrieving access code" maybe?